pkg/controller/dataimportcron-controller.go

```diff
@@ -327,7 +352,16 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 			updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronProgressing, corev1.ConditionFalse, fmt.Sprintf("Import DataVolume phase %s", dvPhase), dvPhase)
 		}
 	} else if pvc != nil {
-		if err = r.updatePvc(ctx, dataImportCron, pvc); err != nil {
+		// TODO: IsPopulated check? wait for corev1.Bound?
+		if err = r.updateSourceObject(ctx, dataImportCron, pvc); err != nil {
+			return res, err
+		}
+		importSucceeded = true
+		if err := r.handleCronFormat(ctx, dataImportCron, format, &dvStorageClass.Name); err != nil {
+			return res, err
+		}
```

Comment:
This block and `case cdiv1.Succeeded` are similar, so maybe refactor or handle them in one place?

Reply:
done